api-consumption/src/main/java/org/zalando/nakadi/service/subscription/SubscriptionRebalancer.java

```diff
@@ -36,11 +40,15 @@
             for (final EventTypePartition requestedPartition : session.getRequestedPartitions()) {
 
                 // find a partition that is requested and assign it to a session that requests it
-                final Partition partition = partitionsLeft.stream()
+                final Optional<Partition> partitionOpt = partitionsLeft.stream()
                         .filter(p -> p.getKey().equals(requestedPartition))
-                        .findFirst()
-                        .orElseThrow(() -> new RebalanceConflictException(
-                                "Two existing sessions request the same partition: " + requestedPartition));
+                        .findFirst();
+                if (!partitionOpt.isPresent()) {
+                    LOG.warn("Two existing sessions request the same partition: " + requestedPartition);
+                    return new Partition[0];
+                }
```

Comment:
This covers only first part of problem - whenever clients have requested partitions. But when there are just more sessions than partitions - It is a different situation - there is a 
```
throw new IllegalArgumentException("Can not rebalance " + itemCount + " onto " + weigths.length);
```
exception later. 

I suggest a different approach do not modify this part of code, but - change `IllegalArgumentException` to `RebalanceConflictException`, For every RebalanceConflictException - do not do rebalance. 

This will allow to have tests for different rebalance problems and will still solve the issue that we have with concurrent sessions creation.